JsonSchema/JsonSchema.cs

```diff
@@ -1,17 +1,102 @@
+[JsonSerializable(typeof(DependentRequiredKeyword))]
+[JsonSerializable(typeof(DependentSchemasKeyword))]
+[JsonSerializable(typeof(DeprecatedKeyword))]
+[JsonSerializable(typeof(DescriptionKeyword))]
+[JsonSerializable(typeof(DynamicAnchorKeyword))]
+[JsonSerializable(typeof(DynamicRefKeyword))]
+[JsonSerializable(typeof(ElseKeyword))]
+[JsonSerializable(typeof(EnumKeyword))]
+[JsonSerializable(typeof(ExamplesKeyword))]
+[JsonSerializable(typeof(ExclusiveMaximumKeyword))]
+[JsonSerializable(typeof(ExclusiveMinimumKeyword))]
+[JsonSerializable(typeof(FormatKeyword))]
+[JsonSerializable(typeof(IdKeyword))]
+[JsonSerializable(typeof(IfKeyword))]
+[JsonSerializable(typeof(ItemsKeyword))]
+[JsonSerializable(typeof(MaxContainsKeyword))]
+[JsonSerializable(typeof(MaximumKeyword))]
+[JsonSerializable(typeof(MaxItemsKeyword))]
+[JsonSerializable(typeof(MaxLengthKeyword))]
+[JsonSerializable(typeof(MaxPropertiesKeyword))]
+[JsonSerializable(typeof(MinContainsKeyword))]
+[JsonSerializable(typeof(MinimumKeyword))]
+[JsonSerializable(typeof(MinItemsKeyword))]
+[JsonSerializable(typeof(MinLengthKeyword))]
+[JsonSerializable(typeof(MinPropertiesKeyword))]
+[JsonSerializable(typeof(MultipleOfKeyword))]
+[JsonSerializable(typeof(NotKeyword))]
+[JsonSerializable(typeof(OneOfKeyword))]
+[JsonSerializable(typeof(PatternKeyword))]
+[JsonSerializable(typeof(PatternPropertiesKeyword))]
+[JsonSerializable(typeof(PrefixItemsKeyword))]
+[JsonSerializable(typeof(PropertiesKeyword))]
+[JsonSerializable(typeof(PropertyDependenciesKeyword))]
+[JsonSerializable(typeof(PropertyNamesKeyword))]
+[JsonSerializable(typeof(ReadOnlyKeyword))]
+[JsonSerializable(typeof(RecursiveAnchorKeyword))]
+[JsonSerializable(typeof(RecursiveRefKeyword))]
+[JsonSerializable(typeof(RefKeyword))]
+[JsonSerializable(typeof(RequiredKeyword))]
+[JsonSerializable(typeof(SchemaKeyword))]
+[JsonSerializable(typeof(ThenKeyword))]
+[JsonSerializable(typeof(TitleKeyword))]
+[JsonSerializable(typeof(TypeKeyword))]
+[JsonSerializable(typeof(UnevaluatedItemsKeyword))]
+[JsonSerializable(typeof(UnevaluatedPropertiesKeyword))]
+[JsonSerializable(typeof(UniqueItemsKeyword))]
+[JsonSerializable(typeof(UnrecognizedKeyword))]
+[JsonSerializable(typeof(VocabularyKeyword))]
+[JsonSerializable(typeof(WriteOnlyKeyword))]
+[JsonSerializable(typeof(JsonNode))]
+[JsonSerializable(typeof(SchemaValueType))]
+[JsonSerializable(typeof(string[]))]
+[JsonSerializable(typeof(Dictionary<string, JsonSchema>))]
+[JsonSerializable(typeof(Dictionary<string, bool>))]
+[JsonSerializable(typeof(List<JsonSchema>))]
+[JsonSerializable(typeof(List<string>))]
+[JsonSerializable(typeof(JsonArray))]
+[JsonSerializable(typeof(Dictionary<string, SchemaOrPropertyList>))]
+[JsonSerializable(typeof(Dictionary<string, List<string>>))]
+[JsonSerializable(typeof(int[]))]
+[JsonSerializable(typeof(Dictionary<string, PropertyDependency>))]
+internal partial class JsonSchemaSerializationContext : JsonSerializerContext
```

Comment:
We had talked about making this public and only carry the public types with another for the internal types.